src/input/mod.rs

```diff
@@ -1625,6 +1632,57 @@ impl State {
             )));
         }
 
+        if event.state() == KeyState::Released {
+            let removed = self
+                .common
+                .atspi_ei
+                .active_virtual_mods
+                .remove(&event.key_code());
+            // If `Caps_Lock` is a virtual modifier, and is in locked state, clear it
+            if removed && handle.modified_sym() == Keysym::Caps_Lock {
+                if (modifiers.serialized.locked & 2) != 0 {
+                    let serial = SERIAL_COUNTER.next_serial();
+                    let time = self.common.clock.now().as_millis();
+                    keyboard.input(
+                        self,
+                        event.key_code(),
+                        KeyState::Pressed,
+                        serial,
+                        time,
+                        |_, _, _| FilterResult::<()>::Forward,
+                    );
+                    let serial = SERIAL_COUNTER.next_serial();
+                    keyboard.input(
+                        self,
+                        event.key_code(),
+                        KeyState::Released,
+                        serial,
+                        time,
+                        |_, _, _| FilterResult::<()>::Forward,
+                    );
+                }
+            }
+        } else if event.state() == KeyState::Pressed
+            && self
+                .common
+                .atspi_ei
+                .virtual_mods
+                .contains(&event.key_code())
+        {
+            self.common
+                .atspi_ei
+                .active_virtual_mods
+                .insert(event.key_code());
+
+            tracing::error!(
```

Comment:
can we downgrade this to at least `info` (probably `debug`)?

Reply:
Oh right, that's a debugging line I added at some point that I didn't mean to commit. (Downgrading to `debug` would be appropriate).